src/block-components/button/edit.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * External dependencies
+ */
+import {
+	AdvancedRangeControl,
+	InspectorStyleControls,
+	PanelAdvancedSettings,
+	BaseControl,
+	AdvancedToolbarControl,
+	ColorPaletteControl,
+	FourRangeControl,
+} from '~stackable/components'
+import { i18n } from 'stackable'
+import {
+	useBlockEl, useBlockHoverState, useAttributeEditHandlers,
+} from '~stackable/hooks'
+
+/**
+ * Internal dependencies
+ */
+import { BorderControls } from '../helpers'
+import { LinkControls } from '../helpers/link'
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n'
+import { Fragment } from '@wordpress/element'
+
+export const Edit = () => {
+	const [ hoverState ] = useBlockHoverState()
+	const blockEl = useBlockEl()
+
+	const {
+		getAttribute,
+	} = useAttributeEditHandlers()
+
+	return (
+		<InspectorStyleControls>
+			<PanelAdvancedSettings
+				title={ __( 'Link', i18n ) }
+				id="link"
+			>
+				<LinkControls attrNameTemplate="link%s" />
+			</PanelAdvancedSettings>
+
+			<PanelAdvancedSettings
+				title={ __( 'Button', i18n ) }
+				id="button"
+			>
+
+				<Fragment>
+					<BaseControl
+						id="stk-button-normal-color-type"
+					>
+						<Fragment>
+							<AdvancedToolbarControl
+								controls={ [
+									{
+										value: '',
+										title: __( 'Single', i18n ),
+									},
+									{
+										value: 'gradient',
+										title: __( 'Gradient', i18n ),
+									},
+								] }
+								attribute="buttonBackgroundColorType"
+								isSmall={ true }
+								fullwidth={ false }
+								hover="all"
```

Comment:
added `hover="all"` or `responsive="all"` to color types make things more complicated, don't use hover and responsive for this. You can check https://github.com/gambitph/Stackable/tree/v3/src/block-components/helpers/backgrounds on how it was used there

Reply:
Please remove the `hover="all"` here